src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
There are some improvements I'd suggest though, to enhance the efficiency, readability of the code:

- Use `.reduce()` to create a map for quick reference of people by their names. This avoids the need to search for a person each time we need them, making the solution more efficient.
- Combine the logic for filtering children into one line using logical AND `&&` and OR `||` operators. This removes the need for nested conditional statements.

```suggestion
  const peopleByName = people.reduce((map, person) => {
    map[person.name] = person;
    return map;
  }, {});
  
  const children = people.filter(p => peopleByName[p.mother] 
    && (!onlyWithSon || (onlyWithSon && p.sex === 'm')));
```